### PR TITLE
feature/AKD-96: Added Azure 'Storage Auto Growth Enabled' plugin and test cases

### PR DESCRIPTION
**Pull request by ali-imran7**

### CODE REVIEW

File: plugins/azure/postgresqlserver/storageAutoGrowthEnabled.spec.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        'sku': {
+            'name': 'B_Gen5_1',
+            'tier': 'Basic',
+            'family': 'Gen5',
+            'capacity': 1
+        },
+        'location': 'eastus',
+        'tags': {},
+        'id': '/subscriptions/dce7d0ad-ebf6-437f-a3b0-28fc0d22117e/resourceGroups/Ali-Recource_Group/providers/Microsoft.DBforPostgreSQL/servers/postgresql-ali',
+        'name': 'postgresql-ali',
```
Comment (AkhtarAmir) [inline plugins/azure/postgresqlserver/storageAutoGrowthEnabled.spec.js:15]: You should never use your own account/subscription Ids
```suggestion
        'id': '/subscriptions/123/resourceGroups/aqua-resource-group/providers/Microsoft.DBforPostgreSQL/servers/server1',
        'name': 'postgresql-ali',
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/azure/postgresqlserver/storageAutoGrowthEnabled.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Storage Auto Growth Enabled',
```
[the remark this answers is not in the document]
Comment (ali-imran7) [reply plugins/azure/postgresqlserver/storageAutoGrowthEnabled.js]: Done as suggested.

File: plugins/azure/postgresqlserver/storageAutoGrowthEnabled.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    category: 'PostgreSQL Server',
+    description: 'Ensures that Storage Auto Growth is enabled for PostgreSQL servers',
+    more_info: 'Storage Auto Growth must be enabled to accommodate the growing data',
```
[the remark this answers is not in the document]
Comment (ali-imran7) [reply plugins/azure/postgresqlserver/storageAutoGrowthEnabled.js]: Done as suggested.

File: plugins/azure/postgresqlserver/storageAutoGrowthEnabled.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Storage Auto Growth Enabled',
+    category: 'PostgreSQL Server',
+    description: 'Ensures that Storage Auto Growth is enabled for PostgreSQL servers',
```
[the remark this answers is not in the document]
Comment (ali-imran7) [reply plugins/azure/postgresqlserver/storageAutoGrowthEnabled.js]: Done as suggested.

File: plugins/azure/postgresqlserver/storageAutoGrowthEnabled.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    more_info: 'Storage Auto Growth must be enabled to accommodate the growing data',
+    recommended_action: 'Enable Storage Auto Growth for PostgreSQL Server',
```
[the remark this answers is not in the document]
Comment (ali-imran7) [reply plugins/azure/postgresqlserver/storageAutoGrowthEnabled.js]: Done as suggested.

File: plugins/azure/postgresqlserver/storageAutoGrowthEnabled.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+                            'Storage Auto Growth is enabled for PostgreSQL Server', location, postgresServer.id);
+                    } else {
+                        helpers.addResult(results, 2,
+                            'Storage Auto Growth is not enabled for PostgreSQL Server', location, postgresServer.id);
+                    }
+                } else {
+                    helpers.addResult(results, 2,
+                        'Storage Auto Growth is not enabled for PostgreSQL Server', location, postgresServer.id);
+                }
+            }
```
[the remark this answers is not in the document]
Comment (ali-imran7) [reply plugins/azure/postgresqlserver/storageAutoGrowthEnabled.js]: Done as suggested.

File: plugins/azure/postgresqlserver/storageAutoGrowthEnabled.spec.js (hunk 1 above)
Comment (ali-imran7) [reply plugins/azure/postgresqlserver/storageAutoGrowthEnabled.spec.js]: Noted and done as suggested.

**Review by giorod3 — APPROVED** [review]

**Review by giorod3 — APPROVED** [review]

**Pull request merged**